### PR TITLE
Connect `LiveViewCoordinator` from `NavigationLink`

### PR DESCRIPTION
**Pull request by carson-katri**
When `<NavigationLink>` was used, the coordinator was never able to connect.

This resolves the issue by handling coordinator connection on forward navigation from the `LiveSessionCoordinator`.

### CODE REVIEW

**Comment by bcardarella** [comment]
I assume this closes #1199 ?

File: Sources/LiveViewNative/LiveView.swift [hunk 1]
```diff
@@ -45,18 +45,15 @@ public macro LiveView<Host: LiveViewHost>(
...
     
     @Environment(\.scenePhase) private var scenePhase
     
     /// Creates a new LiveView attached to the given coordinator.
     ///
     /// - Note: Changing coordinators after the `LiveView` is setup and connected is forbidden.
-    public init(session: LiveSessionCoordinator<R>) {
-        self._session = .init(wrappedValue: session)
-        self.rootCoordinator = session.navigationPath.first!.coordinator
+    public init(session: @autoclosure @escaping () -> LiveSessionCoordinator<R>) {
```
Comment (carson-katri) [inline Sources/LiveViewNative/LiveView.swift:55]: I've made this take a closure instead, which should help with embedding a `LiveView` in an app with state that changes outside.

It's marked `@autoclosure` so the API looks the same as before.

File: Sources/LiveViewNative/LiveView.swift [hunk 2]
```diff
@@ -66,6 +63,10 @@ public struct LiveView<R: RootRegistry>: View {
     public init(url: URL, configuration: LiveSessionConfiguration = .init()) {
         self.init(session: .init(url, config: configuration))
     }
+    
+    private var rootCoordinator: LiveViewCoordinator<R> {
+        session.navigationPath.first!.coordinator
+    }
```
Comment (carson-katri) [inline Sources/LiveViewNative/LiveView.swift:69]: The `rootCoordinator` is now a computed property. This allows `redirect` navigation to be performed from the root of the app.

File: Sources/LiveViewNative/Coordinators/LiveSessionCoordinator.swift [hunk 3]
```diff
@@ -364,19 +364,21 @@ public class LiveSessionCoordinator<R: RootRegistry>: ObservableObject {
...
                 if !navigationPath.isEmpty {
-                    navigationPath[navigationPath.count - 1] = entry
+                    await navigationPath.last?.coordinator.disconnect()
+                    let coordinator = LiveViewCoordinator(session: self, url: redirect.to)
+                    coordinator.url = redirect.to
+                    navigationPath[navigationPath.count - 1] = LiveNavigationEntry(url: redirect.to, coordinator: coordinator)
+                    try await coordinator.connect(domValues: self.domValues, redirect: true)
                 }
             }
-            try await coordinator.connect(domValues: self.domValues, redirect: true)
```
Comment (carson-katri) [inline Sources/LiveViewNative/Coordinators/LiveSessionCoordinator.swift:374]: Connection is now handled from `scan` in `LiveSessionCoordinator.init`.

Only `replace` navigation needs to manage its own reconnection.

**Comment by carson-katri** [comment]
Here's an example of the navigation using a combination of `<NavigationLink>`, `push_navigate` and `push_navigate(replace: true)`:

https://github.com/liveview-native/liveview-client-swiftui/assets/13581484/ed492c22-7551-4279-b22a-b1473d74f83d

**Review by supernintendo — APPROVED** [review]

**Pull request merged**